x-pack/plugins/ml/public/application/explorer/anomaly_context_menu.tsx

```diff
@@ -103,6 +136,183 @@ export const AnomalyContextMenu: FC<AnomalyContextMenuProps> = ({
     return globalTimeRange;
   }, [chartsData.seriesToPlot, globalTimeRange, selectedCells, bounds, interval]);
 
+  const isMaxSeriesToPlotValid =
+    maxSeriesToPlot >= 1 && maxSeriesToPlot <= MAX_ANOMALY_CHARTS_ALLOWED;
+
+  const jobIds = selectedJobs.map(({ id }) => id);
+
+  const getEmbeddableInput = useCallback(
+    (timeRange?: TimeRange) => {
+      // Respect the query and the influencers selected
+      // If no query or filter set, filter out to the lanes the selected cells
+      // And if no selected cells, show everything
+
+      const selectionInfluencers = getSelectionInfluencers(
+        selectedCells,
+        selectedCells?.viewByFieldName!
+      );
+
+      const influencers = selectionInfluencers ?? [];
+      const config = getDefaultEmbeddablePanelConfig(jobIds, queryString);
+      const queryFromSelectedCells = influencers
+        .map((s) => escapeKueryForFieldValuePair(s.fieldName, s.fieldValue))
+        .join(' or ');
+
+      // When adding anomaly charts to Dashboard, we want to respect the Dashboard's time range
+      // so we are not passing the time range here
+      return {
+        ...config,
+        ...(timeRange ? { timeRange } : {}),
+        jobIds,
+        maxSeriesToPlot: maxSeriesToPlot ?? DEFAULT_MAX_SERIES_TO_PLOT,
+        severityThreshold: severity.val,
+        ...((isDefined(queryString) && queryString !== '') ||
+        (queryFromSelectedCells !== undefined && queryFromSelectedCells !== '')
+          ? {
+              query: {
+                query: queryString === '' ? queryFromSelectedCells : queryString,
+                language: SEARCH_QUERY_LANGUAGE.KUERY,
+              } as Query,
+            }
+          : {}),
+      };
+    },
+    [jobIds, maxSeriesToPlot, severity, queryString, selectedCells]
+  );
+
+  const onSaveCallback: SaveModalDashboardProps['onSave'] = useCallback(
+    ({ dashboardId }) => {
+      const stateTransfer = embeddable!.getStateTransfer();
+
+      const embeddableInput: Partial<AnomalyChartsEmbeddableInput> = getEmbeddableInput();
+
+      const state = {
+        input: embeddableInput,
+        type: ANOMALY_EXPLORER_CHARTS_EMBEDDABLE_TYPE,
+      };
+
+      const path = dashboardId === 'new' ? '#/create' : `#/view/${dashboardId}`;
+
+      stateTransfer.navigateToWithEmbeddablePackage('dashboards', {
```

Comment:
If I edit the title, or add a description, neither seem to be getting set in the panel that is added to the dashboard.

Reply:
There seems to be something wrong with the saving of the title and description of the panel, where the title and / or description get lost in between saving and re-opening the dashboard. May not be related to this PR, but I can only reproduce with our embeddable, and not with e.g. a Lens panel.

![dashboard_panel_settings](https://github.com/elastic/kibana/assets/7405507/6fc366df-13db-4b7a-b59c-1ecd8de984c7)

Reply:
This also happens on `main` so please go ahead and create a separate issue for this @darnautov